common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -274,6 +274,7 @@ def partial_update(self, request, *args, **kwargs):
                 )
             )
         support_details = request.data.pop('support_details', [])
+        request.data['marked_unrefundable'] = not entitlement.is_entitlement_refundable()
```

Comment:
Will this overwrite the current value of marked_unrefundable in the request? For example the case where 'marked_unrefundable' in the request is true, but is_entitlement_refundable is also true, seems like it would then set marked_unrefundable to false. 
Should this instead be be 
```
request.data['marked_unrefundable'] = request.data['marked_unrefundable'] or not entitlement.is_entitlement_refundable() 
```

Reply:
Take a look at the method that is being used, In that code she is making the check you mention I think @Mjloturco.

Reply:
I was thinking that the request wouldn't actually include the `marked_unrefundable` value since not all of the necessary information would be present there to know if it should be marked as such, but it could be useful for potential future uses of the endpoint to be able to assign the value explicitly rather than implicitly.